### PR TITLE
Add docker image for style check, add check for docker files placement

### PR DESCRIPTION
**Pull request by qoega**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Added docker image for style check. Added style check that all docker and docker compose files are located in docker directory.

Detailed description / Documentation draft:

We need xmltools and shellcheck for style check.
Needed for #13535. 
Will be enabled in CI after image is already built in master.

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

### CODE REVIEW

File: utils/check-style/check-duplicate-includes.sh [hunk 1]
```diff
@@ -1,6 +0,0 @@
-#!/usr/bin/env bash
```
Comment (akuzm) [inline utils/check-style/check-duplicate-includes.sh:1]: Why is it deleted, did you intend to move it to docker dir?

**Review by akuzm — APPROVED** [review]

File: utils/check-style/check-duplicate-includes.sh (hunk 1 above)
Comment (qoega) [reply utils/check-style/check-duplicate-includes.sh:1]: Actually is was an accident and actual changes are not there..

**Comment by qoega** [comment]
Style check will be green after #13647

**Review by alexey-milovidov — APPROVED** [review]

**Comment by akuzm** [comment]
Container build OK

**Pull request merged**